xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/TestUtils.java

```diff
@@ -1610,6 +1630,25 @@ public void attachFile(EntityReference reference, Object is, boolean failIfExist
         rest().attachFile(reference, is, failIfExists);
     }
 
+    /**
+     * @since 12.2RC1
+     */
+    public void attachFile(EntityReference pageReference, String name, InputStream is, boolean failIfExists,
+        UsernamePasswordCredentials credentials) throws Exception
+    {
+        UsernamePasswordCredentials currentCredentials = getDefaultCredentials();
+        EntityReference reference = new EntityReference(name, EntityType.ATTACHMENT, pageReference);
+
+        try {
+            if (credentials != null) {
+                setDefaultCredentials(credentials);
+            }
+            attachFile(reference, is, failIfExists);
+        } finally {
+            setDefaultCredentials(currentCredentials);
```

Comment:
I wouldn't call this method if the credentials have not been changed, will be more performant.

Reply:
setDefaultCredentials does not really cost anything